include/kodlab_mjbots_sdk/rotations.h

```diff
@@ -0,0 +1,603 @@
+ *       and verification for robotics" (2019)
+ * @link https://github.com/RobotLocomotion/drake/blob/5022e3832e90f988f1b6d3015b2794ea26e527b1/math/rotation_matrix.h#L970 @endlink
+ *
+ * @tparam Scalar[optional] numeric type (default float)
+ * @param quaternion
+ * @param two_over_norm_squared
+ * @return Eigen::Matrix3<Scalar>
+ */
+template<typename Scalar = float>
+inline Eigen::Matrix3<Scalar> QuaternionToRotationMatrix(
+    const Eigen::Quaternion<Scalar> &quaternion,
+    const Scalar &two_over_norm_squared)
+{
+  const Scalar w = quaternion.w();
+  const Scalar x = quaternion.x();
+  const Scalar y = quaternion.y();
+  const Scalar z = quaternion.z();
+  const Scalar sx = two_over_norm_squared * x; // scaled x-value.
+  const Scalar sy = two_over_norm_squared * y; // scaled y-value.
+  const Scalar sz = two_over_norm_squared * z; // scaled z-value.
+  const Scalar swx = sx * w;
+  const Scalar swy = sy * w;
+  const Scalar swz = sz * w;
+  const Scalar sxx = sx * x;
+  const Scalar sxy = sy * x;
+  const Scalar sxz = sz * x;
+  const Scalar syy = sy * y;
+  const Scalar syz = sz * y;
+  const Scalar szz = sz * z;
+
+  Eigen::Matrix3<Scalar> m;
+  m.coeffRef(0, 0) = Scalar(1) - syy - szz;
+  m.coeffRef(0, 1) = sxy - swz;
+  m.coeffRef(0, 2) = sxz + swy;
+  m.coeffRef(1, 0) = sxy + swz;
+  m.coeffRef(1, 1) = Scalar(1) - sxx - szz;
+  m.coeffRef(1, 2) = syz - swx;
+  m.coeffRef(2, 0) = sxz - swy;
+  m.coeffRef(2, 1) = syz + swx;
+  m.coeffRef(2, 2) = Scalar(1) - sxx - syy;
+
+  return m;
+}
+
+/**
+ * @brief Constructs a 3x3 rotation matrix from a normalized quaternion.
+ * @note See the cited Drake code for details
+ * @cite Russ Tedrake and the Drake Development Team, "Drake: Model-based design
+ *       and verification for robotics" (2019)
+ * @link https://github.com/RobotLocomotion/drake/blob/5022e3832e90f988f1b6d3015b2794ea26e527b1/math/rotation_matrix.h#L970 @endlink
+ *
+ * @tparam Scalar[optional] numeric type (default float)
+ * @param unit_quaternion
+ * @return Eigen::Matrix3<Scalar>
+ */
+template<typename Scalar = float>
+inline Eigen::Matrix3<Scalar> QuaternionToRotationMatrix(
+    const Eigen::Quaternion<Scalar> &unit_quaternion)
+{
+  return QuaternionToRotationMatrix(unit_quaternion, Scalar(2));
+}
+
+/**
+ * @brief Constructs a Extrinsic X-Y-Z euler angle representation from a 
+ *        normalized quaternion and a rotation matrix
+ * @note See the cited Drake code for details
+ * @cite Russ Tedrake and the Drake Development Team, "Drake: Model-based design
+ *       and verification for robotics" (2019)
+ * @link https://github.com/RobotLocomotion/drake/blob/5022e3832e90f988f1b6d3015b2794ea26e527b1/math/roll_pitch_yaw.cc#L112 @endlink
+ *
+ * @tparam Scalar[optional] numeric type (default float)
+ * @param quaternion
+ * @param R
+ * @return EulerAngles<Scalar>
+ */
+template<typename Scalar = float>
+EulerAngles<Scalar> QuaternionAndRotationMatrixToEulerAngles(
```

Comment:
These last two functions should be renamed to match above. Namely
```suggestion
EulerAngles<Scalar> QuaternionAndRotationMatrixToDefaultEulerAngles(
```

Reply:
Resolved in d3d867197e2aa610a80dc502b3811b3b1d9302de